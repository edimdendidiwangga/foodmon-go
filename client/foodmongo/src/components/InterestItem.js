import React, { Component } from 'react';
import { View, Image, Text, StyleSheet, TouchableOpacity, Dimensions } from 'react-native';
import { connect } from 'react-redux';

import { addInterest, removeInterest } from '../actions';


class InterestItem extends Component {
  constructor(props) {
    super(props);
    const { width, height } = Dimensions.get('window');
    this.state = {
      width,
      height,
    };

    this.styles = StyleSheet.create({
      backdropView: {
        height: this.state.width / 3,
        width: this.state.width / 3,
        backgroundColor: 'rgba(0,0,0,0)',
        borderWidth: 1,
        borderStyle: 'solid',
        borderColor: 'white',
      },
      headline: {
        fontSize: 20,
        textAlign: 'center',
        backgroundColor: 'rgba(0,0,0,0)',
        color: 'white',
        paddingTop: 10,
      },
      headlineBlack: {
        fontSize: 20,
        textAlign: 'center',
        backgroundColor: 'rgba(0,0,0,0)',
        color: 'black',
      },
    });
  }


  handleInterest() {
    if(this.props.user.interestArr) {
      for (let i = 0; i < this.props.user.interestArr.length; i++) {
        if (this.props.user.interestArr[i]._id === this.props.cuisine._id){
          return (
            <TouchableOpacity
              onPress={() => { this.props.removeInterest(this.props.cuisine, this.props.user); }}
            >
              <Image
                source={{ uri: `http://loremflickr.com/g/300/300/${this.props.cuisine.cuisine_name},food/all` }}
                style={this.styles.backdropView}
              >
                <View>
                  <Text style={this.styles.headline}>
                    {this.props.cuisine.cuisine_name} selected
                  </Text>
                </View>
              </Image>
            </TouchableOpacity>
          )
        }
      }
    }

    // this.props.user.interestArr.map(interest=)
    return (
      <TouchableOpacity
        onPress={() => { this.props.addInterest(this.props.cuisine, this.props.user); }}
      >
        <Image
          source={{ uri: `http://loremflickr.com/300/300/${this.props.cuisine.cuisine_name},food/all` }}
          style={this.styles.backdropView}
        >
          <View>
            <Text style={this.styles.headline}>
              {this.props.cuisine.cuisine_name}
            </Text>
          </View>
        </Image>
      </TouchableOpacity>
    );
  }
  render() {
    return (
      <View>
        {this.handleInterest()}
      </View>
    );
  }
}

// const InterestItem = props => (
//   <Image style={styles.backdropView} source={{ uri: `http://loremflickr.com/320/240/${props.cuisine.cuisine_name},food/all` }}>
//     <View>
//       <Text style={styles.headline}>
//         {props.cuisine.cuisine_name}
//       </Text>
//     </View>
//   </Image>
// );

const mapStateToProps = state => ({
  user: state.UserReducer,
});

const mapDispatchToProps = dispatch => ({
  addInterest: (interest, user) => dispatch(addInterest(interest, user)),
  removeInterest: (interest, user) => dispatch(removeInterest(interest, user)),
});

export default connect(mapStateToProps, mapDispatchToProps)(InterestItem);
