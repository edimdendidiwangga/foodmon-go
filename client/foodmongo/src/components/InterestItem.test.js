import renderer from 'react-test-renderer';
import React from 'react';

import { InterestItem } from './InterestItem'

describe('InterestItem', () => {

  it('renders and match to snapshot', () => {
    const interests = [
      {
        _id: '593a4f628767cb7b789e1ffc',
        cuisine_name: 'Aceh',
        cuisine_id: '237',
        __v: 0,
      },
      {
        _id: '593a4f768767cb7b789e1ffd',
        cuisine_name: 'American',
        cuisine_id: '1',
        __v: 0,
      },
    ];

    const cuisine = {
      _id: '593a4f628767cb7b789e1ffc',
      cuisine_name: 'Aceh',
      cuisine_id: '237',
      __v: 0
    };

    const tree = renderer.create(<InterestItem user={{ interestArr: interests }} cuisine={cuisine} addInterest={(interest, user) => {}} removeInterest={(interest, user) => {}} />);
    expect(tree).toMatchSnapshot();
  });
});
