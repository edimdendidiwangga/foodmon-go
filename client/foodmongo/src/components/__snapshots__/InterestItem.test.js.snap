// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterestItem renders and match to snapshot 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      source={
        Object {
          "uri": "http://loremflickr.com/g/300/300/Aceh,food/all",
        }
      }
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderColor": "white",
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 250,
          "width": 250,
        }
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0)",
              "color": "white",
              "fontSize": 20,
              "paddingTop": 10,
              "textAlign": "center",
            }
          }
        >
          Aceh
           selected
        </Text>
      </View>
    </Image>
  </View>
</View>
`;
